tests/models/test_sasrec.py

```diff
@@ -582,3 +593,337 @@ def test_get_dataloader_recommend(
+                    ],
+                    columns=["id", "feature", "value"],
+                ),
+                (),
+                False,
+            ),
+            (
+                pd.DataFrame(
+                    [
+                        [11, 1, 1],
+                        [12, 1, 2],
+                        [13, 1, 3],
+                        [14, 2, 1],
+                        [15, 2, 2],
+                        [17, 2, 3],
+                    ],
+                    columns=[Columns.Item, "f1", "f2"],
+                ),
+                ["f1", "f2"],
+                True,
+            ),
+        ),
+    )
+    def test_when_cat_item_features_is_none(
+        self,
+        item_features: tp.Optional[pd.DataFrame],
+        cat_item_features: tp.Iterable[str],
+        make_dense_item_features: bool,
+    ) -> None:
+        ds = Dataset.construct(
+            INTERACTIONS,
+            item_features_df=item_features,
+            cat_item_features=cat_item_features,
+            make_dense_item_features=make_dense_item_features,
+        )
+        cat_features_item_net = CatFeaturesItemNet.from_dataset(ds, n_factors=10, dropout_rate=0.5)
+        assert cat_features_item_net is None
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+class TestItemNetConstructor:
+    def setup_method(self) -> None:
+        self._seed_everything()
+
+    def _seed_everything(self) -> None:
+        torch.use_deterministic_algorithms(True)
+        seed_everything(32, workers=True)
+
+    @pytest.fixture
+    def dataset_item_features(self) -> Dataset:
+        item_features = pd.DataFrame(
+            [
+                [11, "f1", "f1val1"],
+                [11, "f2", "f2val1"],
+                [12, "f1", "f1val1"],
+                [12, "f2", "f2val2"],
+                [13, "f1", "f1val1"],
+                [13, "f2", "f2val3"],
+                [14, "f1", "f1val2"],
+                [14, "f2", "f2val1"],
+                [15, "f1", "f1val2"],
+                [15, "f2", "f2val2"],
+                [17, "f1", "f1val2"],
+                [17, "f2", "f2val3"],
+                [16, "f1", "f1val2"],
+                [16, "f2", "f2val3"],
+                [11, "f3", 0],
+                [12, "f3", 1],
+                [13, "f3", 2],
+                [14, "f3", 3],
+                [15, "f3", 4],
+                [17, "f3", 5],
+                [16, "f3", 6],
+            ],
+            columns=["id", "feature", "value"],
+        )
+        ds = Dataset.construct(
+            INTERACTIONS,
+            item_features_df=item_features,
+            cat_item_features=["f1", "f2"],
+        )
+        return ds
+
+    def test_catalogue(self) -> None:
+        item_net = ItemNetConstructor.from_dataset(
+            DATASET, n_factors=10, dropout_rate=0.5, item_net_block_types=(IdEmbeddingsItemNet,)
+        )
+        expected_feature_catalogue = torch.arange(0, item_net.n_items)
```

Comment:
You are just reusing exactly the same code that is in the class method. This doesn't make any sense for a test. You can either check it from number of items in dataset. Or just drop this test)

Reply:
Fixed